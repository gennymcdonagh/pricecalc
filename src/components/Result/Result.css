.result-list {
  list-style-type: none;
  padding: 0;
}

.result-list li {
  padding: 5px;
  display: flex;
  justify-content: center;
}

.result-list__label {
  width: 50%;
  text-align: right;
  margin-right: 10px;
}

.result-list__value {
  width: 50%;
  text-align: left;
}