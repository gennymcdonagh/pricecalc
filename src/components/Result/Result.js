import React from 'react';
import './Result.css';
import PropTypes from 'prop-types';

export const Result = (props) => {
    const { resultData } = props;
    let html = [];

    resultData.forEach(s => {
      html.push(
        <h3 key={s.sectionName}>{s.sectionName}</h3>
      );

      for (const prop in s.sectionData) {
        html.push(
          <li key={`${s.sectionName} ${prop}`}>
            <span className="result-list__label">{prop}: </span>
            <span className="result-list__value">${s.sectionData[prop]}</span>
          </li>
        );
      }
    })
    
    return (
      <div className="result-list">
        {html}
      </div>
    )
};

Result.propTypes = {
  resultData: PropTypes.arrayOf(
    PropTypes.shape({
      sectionName: PropTypes.string,
      sectionData: PropTypes.objectOf(PropTypes.number)
    })
  )
}

export default Result;