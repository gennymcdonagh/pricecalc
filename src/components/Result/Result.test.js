import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Result from './Result';

it('renders correctly', () => {
  const result = renderer.create(<Result data={{"key1":"val1","key2":"val2"}}/>).toJSON();
  expect(result).toMatchSnapshot();
})
