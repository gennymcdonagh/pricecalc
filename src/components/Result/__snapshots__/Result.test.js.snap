// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="result-list"
>
  <li>
    <span
      className="result-list__label"
    >
      key1
      : 
    </span>
    <span
      className="result-list__value"
    >
      $
      val1
    </span>
  </li>
  <li>
    <span
      className="result-list__label"
    >
      key2
      : 
    </span>
    <span
      className="result-list__value"
    >
      $
      val2
    </span>
  </li>
</ul>
`;
